src/components/molecules/ImageInput/ImageInput.tsx

```diff
@@ -24,94 +24,82 @@ interface ImageInputProps extends ContainerClassName {
+    async (event: ChangeEvent<HTMLInputElement>) => {
+      const [url, compressedFile] = await handleFileInputChange(event);
+      if (!compressedFile || !url) return;
 
-        setValue(name, [compressedFile], false);
+      setValue(name, [compressedFile], false);
 
-        if (remoteUrlInputName) {
-          setValue(remoteUrlInputName, url, true);
-        }
-      },
-      [handleFileInputChange, name, remoteUrlInputName, setValue]
-    );
+      if (remoteUrlInputName) {
+        setValue(remoteUrlInputName, url, true);
+      }
+    },
+    [handleFileInputChange, name, remoteUrlInputName, setValue]
+  );
 
-    return (
-      <>
-        <div
-          className={classNames(
-            `image-input default-container`,
-            containerClassName,
-            {
-              disabled: loading,
-            }
-          )}
-        >
-          {imageUrl ? (
-            <img
-              className={`default-image ${imageClassName}`}
-              src={imageUrl}
-              alt="upload"
-            />
-          ) : (
-            <div className="centered-flex empty">
-              <h6 className=" text" style={{ fontWeight: "lighter" }}>
-                Click to upload an image
-              </h6>
-            </div>
-          )}
-          <input
-            disabled={disabled || loading}
-            type="file"
-            accept={ACCEPTED_IMAGE_TYPES}
-            onChange={handleFileInputChangeWrapper}
-            className="default-input"
+  return (
+    <>
+      <div
+        className={classNames(
+          `image-input default-container`,
+          containerClassName,
+          {
+            disabled: loading,
+          }
+        )}
+      >
+        {imageUrl ? (
+          <img
+            className={`default-image ${imageClassName}`}
+            src={imageUrl}
+            alt="upload"
           />
-          {loading && <ImageOverlay>processing...</ImageOverlay>}
-          {remoteUrlInputName && (
-            <input
-              type="hidden"
-              ref={ref}
-              name={remoteUrlInputName}
-              value={remoteImageUrl}
-            />
-          )}
-        </div>
-        {errorMessage && <span className="input-error">{errorMessage}</span>}
-      </>
-    );
-  }
-);
-
-ImageInput.displayName = "ImageInput";
+        ) : (
+          <div className="centered-flex empty">
+            <h6 className=" text" style={{ fontWeight: "lighter" }}>
+              Click to upload an image
+            </h6>
+          </div>
+        )}
+        <input
+          disabled={disabled || loading}
+          type="file"
+          accept={ACCEPTED_IMAGE_TYPES}
+          onChange={handleFileInputChangeWrapper}
+          className="default-input"
+        />
+        {loading && <ImageOverlay>processing...</ImageOverlay>}
+        {remoteUrlInputName && (
+          <input type="hidden" ref={register} name={remoteUrlInputName} />
```

Comment:
This file looks like a lot of changes but that's mostly indentaion. 
The useful one is this:
before:
```typesctipt
<input
              type="hidden"
              ref={ref}
              name={remoteUrlInputName}
              value={remoteImageUrl}
            />
```
after
```typescript
<input type="hidden" ref={register} name={remoteUrlInputName} />
```

So I removed the `forwardingRef` from this component and passed correct `register` ref to the `input`

Reply:
> So I removed the forwardingRef from this component and passed correct register ref to the input

I'm still not used to `forwardingRef` but I did come into issues just trying to get the `register` function from the outside, especially when the validation options should be dictated by parent component where the form resides.

Are you sure this will not cause issues down the road?

Reply:
Do not now for any possible issues, but currently, it's working as intended.